runtime/src/iree/base/internal/synchronization_test.cc

```diff
@@ -132,9 +132,10 @@ void TestMutexExclusiveAccessTryLock() {
 TEST(MutexTest, Lifetime) {
   iree_mutex_t mutex;
   iree_mutex_initialize(&mutex);
-  bool did_lock = iree_mutex_try_lock(&mutex);
-  EXPECT_TRUE(did_lock);
-  if (did_lock) iree_mutex_unlock(&mutex);
+  while (!iree_mutex_try_lock(&mutex)) {
+    // NOTE: some platforms (arm64) can fail tries spuriously so we retry.
```

Comment:
Nit: implementation != interface. This comment is about the underlying implementation and is correct about that, but this test code shouldn't have to know. More to the point would be something like the title of https://github.com/openxla/iree/pull/15636: "functions with _try_ in their name may fail spuriously".